src/components/VHover/VHover.ts

```diff
@@ -0,0 +1,73 @@
+// Mixins
+import Delayable from '../../mixins/delayable'
+import Toggleable from '../../mixins/toggleable'
+
+// Utilities
+import mixins from '../../util/mixins'
+import { consoleWarn } from '../../util/console'
+
+// Types
+import { VNode } from 'vue'
+
+export default mixins(
+  Delayable,
+  Toggleable
+).extend({
+  name: 'v-hover',
+
+  props: {
+    disabled: {
+      type: Boolean,
+      default: false
+    },
+    value: {
+      type: Boolean,
+      default: undefined
+    }
+  },
+
+  methods: {
+    __onMouseEnter () {
+      this.isActive = true
+    },
+    __onMouseLeave () {
+      this.isActive = false
+    }
+  },
+
+  render (): any {
+    if (!this.$scopedSlots.default && this.value === undefined) {
+      consoleWarn('v-hover is missing a default scopedSlot or bound value', this)
+
+      return null
+    }
+
+    let element = null
+
+    // TODO: types are wrong - https://github.com/vuejs/vue/pull/8644
+    if (this.$scopedSlots.default) {
+      element = this.$scopedSlots.default({ hover: this.isActive }) as any as VNode
+    } else if (this.$slots.default.length === 1) {
+      element = this.$slots.default[0]
+    }
+
+    if (!element || typeof element === 'string' || Array.isArray(element)) {
+      consoleWarn('v-hover should only contain a single element', this)
+
+      return element
+    }
+
+    if (!this.disabled) {
+      element.data!.on = element.data!.on || {}
+
+      element.data!.on!.mouseenter = () => {
```

Comment:
Does this overwrite existing listeners? Might need to merge them: https://github.com/vuejs/vue/blob/25a688ee314d82b584ef0f192d4016b75664a616/src/core/instance/render-helpers/bind-object-listeners.js#L13